editor.planx.uk/src/@planx/components/shared/Schema/SchemaFields.tsx

```diff
@@ -0,0 +1,57 @@
+import { FormikProps } from "formik";
+import React from "react";
+import InputRow from "ui/shared/InputRow";
+
+import { ChecklistFieldInput, DateFieldInput, NumberFieldInput, RadioFieldInput, SelectFieldInput, TextFieldInput } from "./Fields";
+import { Field, Schema, SchemaData } from "./model";
+
+type InputFieldProps = {
+  activeIndex: number;
+  formik: FormikProps<SchemaData>;
+} & Field;
+
+/**
+* Controller to return correct user input for field in schema
+*/
+const InputField: React.FC<InputFieldProps> = (props) => {
+  switch (props.type) {
+    case "text":
+      return <TextFieldInput {...props} />;
+    case "number":
+      return <NumberFieldInput {...props} />;
+    case "question":
+      if (props.data.options.length === 2) {
+        return <RadioFieldInput {...props} />;
+      }
+      return <SelectFieldInput {...props} />;
+    case "checklist":
+      return <ChecklistFieldInput {...props} />;
+    case "date":
+      return <DateFieldInput {...props} />;
+    default:
+      return null;
```

Comment:
To do (follow up PR): Add an exhaustive check on a few of these switch statements, or take a look at https://typescript-eslint.io/rules/switch-exhaustiveness-check/ as an option.